### PR TITLE
Add drush deploy:hook-skip (#4738)

### PR DESCRIPTION
**Pull request by bircher**
PR for #4738

I am open to name it differently, have it ask for confirmation, expand the documentation etc...

### CODE REVIEW

File: src/Drupal/Commands/core/DeployHookCommands.php [hunk 1]
```diff
@@ -153,4 +153,22 @@ public function run()
...
         return $success ? self::EXIT_SUCCESS : self::EXIT_FAILURE;
     }
+
+    /**
+     * Mark all deploy hooks as having run.
+     *
+     * @usage deploy:hook-skip
+     *   Skip all pending deploy hooks.
+     *
+     * @command deploy:hook-skip
```
Comment (weitzman) [inline src/Drupal/Commands/core/DeployHookCommands.php:163]: How about the name `deploy:mark-complete`? Change `@usage` description and method names to match.

File: tests/functional/DeployHookTest.php [hunk 2]
```diff
@@ -55,4 +55,45 @@ public function testDeployHooks()
...
+    {
+        "module": "woot",
+        "hook": "failing",
+        "description": "Failing deploy hook."
+    }
+]
+JSON;
+        // Check pending deploy hooks.
+        $this->drush('deploy:hook-status', [], $options, null, null, self::EXIT_SUCCESS);
+        $this->assertEquals(json_decode($hooks), json_decode($this->getOutput()));
```
Comment (weitzman) [inline tests/functional/DeployHookTest.php:89]: Simplify with `$this->getOutputFromJSON()`

**Review by weitzman — CHANGES_REQUESTED** [review]
Lets add a paragraph about this in the deploy topic. Please mention the use case(s) where it is helpful.

**Comment by Leksat** [comment]
@bircher are you able to continue work on this PR? It's almost perfect :)

File: src/Drupal/Commands/core/DeployHookCommands.php (hunk 1 above)
Comment (bircher) [reply src/Drupal/Commands/core/DeployHookCommands.php]: great suggestion. naming is hard

File: tests/functional/DeployHookTest.php (hunk 2 above)
Comment (bircher) [reply tests/functional/DeployHookTest.php]: oh of course! how could I have overlooked that... makes things much simpler!

**Comment by bircher** [comment]
Sorry for neglecting the PR! I think I need to tune the notification settings a bit.

File: src/Drupal/Commands/core/DeployHookCommands.php [hunk 3]
```diff
@@ -253,4 +253,22 @@ public function updateFinished($success, $results, $operations)
     {
         // In theory there is nothing to do here.
     }
+
+    /**
+     * Mark all deploy hooks as having run.
+     *
+     * @usage deploy:mark-complete
+     *   Skip all pending deploy hooks and mork them as completed.
```
Comment (weitzman) [inline src/Drupal/Commands/core/DeployHookCommands.php:261]:
```suggestion
     *   Skip all pending deploy hooks and mark them as complete.
```

File: src/Drupal/Commands/core/DeployHookCommands.php [hunk 4]
```diff
@@ -253,4 +253,22 @@ public function updateFinished($success, $results, $operations)
...
+     *
+     * @command deploy:mark-complete
+     * @topics docs:deploy
+     */
+    public function markComplete()
+    {
+        $pending = self::getRegistry()->getPendingUpdateFunctions();
+        self::getRegistry()->registerInvokedUpdates($pending);
+
+        $this->logger()->success(dt('Marked %count pending deploy hooks as completed.', ['%count' => count($pending)]));
```
Comment (weitzman) [inline src/Drupal/Commands/core/DeployHookCommands.php:271]:
```suggestion
        $this->logger()->success(dt('Marked %count pending deploy hooks as complete.', ['%count' => count($pending)]));
```

**Review by weitzman — APPROVED** [review]

**Pull request merged**